ReactiveCocoa/Swift/Signal.swift

```diff
@@ -883,6 +898,17 @@ public func combineLatest<A, B, C, D, E, F, G, H, I, J, Error>(a: Signal<A, Erro
 		|> map(repack)
 }
 
+/// Combines the values of all the given signals, in the manner described by
+/// `combineLatestWith`.
+public func combineLatest<T, Error>(signals: [Signal<T, Error>]) -> Signal<[T], Error> {
+	if let first = signals.first {
+		let initial = first |> map { [$0] }
+		return signals[1 ..< signals.count].reduce(initial) { $0 |> combineLatestWith($1) }
+	}
+	
+	return Signal.never
```

Comment:
I'm inclined to say that passing in an empty sequence should be a precondition failure, but I could be convinced otherwise.

@larcus94 Thoughts?